### PR TITLE
Channels listing and details page 

### PR DESCRIPTION
**Pull request by mishramonalisha76**
### Ticket Number

- [1856](#1856 )

### Description

- **Problem/Feature**:

### Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other (please describe):

### Checklist

- [ ] **Quick PR**: Is this a quick PR? Can be approved before finishing a coffee.
  - [ ] Quick PR label added
- [ ] **Not Merge Ready**: Is this PR dependent on some other PR/tasks and not ready to be merged right now.
  - [ ] DO NOT Merge PR label added

### Frontend Guidelines

- [x] Followed frontend guidelines as per the [Guidelines Notion Doc](https://www.notion.so/pushprotocol/Frontend-dApp-Guidelines-1d7806ae3d9e4569a340b563dcd0536c)

### Build & Testing

- [x] No errors in the build terminal
- [x] Engineer has tested the changes on their local environment
- [x] Engineer has tested the changes on deploy preview

### Screenshots/Video with Explanation

- **Before:** Explain the previous behavior

- **After:** What's changed now

### Review & Approvals

- [ ] Self-review completed
- [ ] Code review by at least one other engineer
- [ ] Documentation updates if applicable

### CODE REVIEW

**Comment by rohitmalhotra1420** [comment]
@mishramonalisha76 

- Skeleton state is not working correctly.
- Spacing between notifications is not correct.
- Last notification is not visible 
- Infinite scroll loader not visible for the channel is not visible
- last channel icon is getting truncated.
- When i load the channel detail page, initially broken channel img with some data is visible and after that skeleton is visible.
- The channel icon will also have a border radius and border.
- Channel icon is shrinking in mobile view.

File: src/modules/channelDetails/components/ChannelDetail.tsx [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+            <Box display={{ dp: 'inline', ml: 'none' }}>
+              <ChannelDetailSubscribe channel={channel} />
+            </Box>
+          </Box>
+        </Box>
+
+        <Skeleton isLoading={isLoading}>
+          <Box>
+            <Text
+              display={{ dp: 'inline', ml: 'none' }}
```
Comment (rohitmalhotra1420) [inline src/modules/channelDetails/components/ChannelDetail.tsx:227]:
```suggestion
              display={{ initial: 'inline', ml: 'none' }}
```

Fix this everywhere in this PR.

File: src/modules/channelDetails/components/ChannelDetail.tsx [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+                {' '}
+                {`Show ${!showMore ? 'More' : 'Less'}`}
+              </Text>
+              <Text
+                display={{ dp: 'none', ml: 'inline' }}
+                color="text-brand-medium"
+                as="span"
+              >
+                {' '}
+                {`Show ${!showMore ? 'More' : 'Less'}`}
```
Comment (rohitmalhotra1420) [inline src/modules/channelDetails/components/ChannelDetail.tsx:261]: This should only be visible if the char length is above a certain limit.

File: src/modules/channelDetails/components/ChannelDetail.tsx [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+
+        <Skeleton isLoading={isLoading}>
+          <Box>
+            <Text
+              display={{ dp: 'inline', ml: 'none' }}
+              variant="bs-regular"
+              color="text-tertiary"
+              as="span"
+            >
+              {!showMore ? (channel?.info || '').substring(0, 250) : channel?.info || ''}
```
Comment (rohitmalhotra1420) [inline src/modules/channelDetails/components/ChannelDetail.tsx:232]: Extract this out in a const

File: src/modules/channelDetails/components/ChannelDetail.tsx [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+
+import { getChannelTutorialDetails } from '../ChannelDetails.utils';
+
+export type ChannelDetailProps = { channel: ChannelDetails; isLoading: boolean };
+const ChannelDetail: FC<ChannelDetailProps> = ({ channel, isLoading }) => {
+  const [showMore, setShowMore] = useState<boolean>(false);
+  const navigate = useNavigate();
+
+  //fetch channel category
+  //notification component doesnot update in sdk
```
Comment (rohitmalhotra1420) [inline src/modules/channelDetails/components/ChannelDetail.tsx:27]: Remove it.

File: src/modules/channelDetails/ChannelDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  const { data: channelDetail } = useGetChannelDetails(id || '');
+
+  const channelsList = isLoading ? Array(10).fill(0) : data?.pages.flatMap((page) => page.channels) || [];
+  const hasMoreData = !isFetchingNextPage && hasNextPage;
+
+  const selectedChannel = channelsList?.find((channel) => channel?.channel === selectedChannelId) || channelDetail;
+
+  useEffect(() => {
+    if (!isAddress(id || '')) navigate(APP_PATHS.Channels);
+  }, [id]);
```
Comment (rohitmalhotra1420) [inline src/modules/channelDetails/ChannelDetails.tsx:31]: Add docstring comments for this hook.

File: src/modules/channelDetails/ChannelDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+  return (
+    channelsList &&
+    channelsList.length && (
```
Comment (rohitmalhotra1420) [inline src/modules/channelDetails/ChannelDetails.tsx:35]:
```suggestion
    channelsList?.length && (
```

**Review by rohitmalhotra1420 — CHANGES_REQUESTED** [review]
@mishramonalisha76 to fix the PR comments.

File: src/queries/hooks/channels/useGetChannelsList.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+import { getChannelsList } from '../../services';
+
+import { ChannelsListModelledResponse, ChannelListParams } from '../../types';
+
+export type UseChannelSearchProps = {
+  order: ChannelListParams['order'];
+  pageSize: ChannelListParams['pageSize'];
+  chain: ChannelListParams['chain'];
+  tag: ChannelListParams['tag'];
+  sort: ChannelListParams['sort'];
```
Comment (mishramonalisha76) [inline src/queries/hooks/channels/useGetChannelsList.ts:13]: order, tag, chain and sort can be optional

**Comment by mishramonalisha76** [comment]
The font weight of the channel title in channel listing page doesnot match the one in figma (700 instead of 500 ) @rohitmalhotra1420

**Review by abhishek-01k — DISMISSED** [review]

File: src/modules/channels/Channels.utils.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import { channelCategoriesMap, envUtil } from 'common';
+import { ChannelDetails } from 'queries';
+import { Filters } from './hooks/useChannelsFilters';
+import { AllCategories, channelFilterList } from './Channels.constants';
+
+export const showFrontendChannels = (filters: Filters) => {
+  return envUtil.isProd && filters?.category && filters?.category != AllCategories;
+};
+
+export const filterFrontendChannels = (channels: ChannelDetails[], filter: Filters): Array<string> => {
```
Comment (rohitmalhotra1420) [inline src/modules/channels/Channels.utils.ts]:
```suggestion
export const getSuggestedChannels = (channels: ChannelDetails[], filter: Filters): Array<string> => {
```

File: src/modules/channels/Channels.utils.ts [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import { channelCategoriesMap, envUtil } from 'common';
+import { ChannelDetails } from 'queries';
+import { Filters } from './hooks/useChannelsFilters';
+import { AllCategories, channelFilterList } from './Channels.constants';
+
+export const showFrontendChannels = (filters: Filters) => {
```
Comment (rohitmalhotra1420) [inline src/modules/channels/Channels.utils.ts]:
```suggestion
export const showSuggestedChannels = (filters: Filters) => {
```

File: src/modules/channels/Channels.tsx [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+  });
+
+  const channels =
+    loadingChannels || searchingChannels
+      ? Array(9).fill(0)
+      : (filters.search ? searchList : channelList)?.pages
+          .flatMap((page) => page.channels)
+          .filter((channel) => !channelFilterList.includes(channel.channel)) || [];
+
+  const filteredFrontendChannels = filterFrontendChannels(channels, filters);
```
Comment (rohitmalhotra1420) [inline src/modules/channels/Channels.tsx]:
```suggestion
  const suggestedChannels = getSuggestedChannels(channels, filters);
```

File: src/modules/channels/Channels.tsx [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+              size={48}
+              color="icon-tertiary"
+            />
+            <Text variant="h5-bold">No channels to display</Text>
+          </Box>
+        ) : (
+          <AllChannelList
+            channels={channels}
+            hasMoreData={hasMoreData}
+            frontendChannels={filteredFrontendChannels}
```
Comment (rohitmalhotra1420) [inline src/modules/channels/Channels.tsx]:
```suggestion
            suggestedChannels={suggestedChannels}
```

File: src/modules/channels/components/AllChannelsList.tsx [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+import { AllChannelsListItem } from './AllChannelsListItem';
+import { FrontendChannelListItem } from './FrontendChannelListItem';
+
+export type AllChannelListProps = {
+  channels: ChannelDetails[];
+  hasMoreData: boolean;
+  isLoading: boolean;
+  isLoadingNextPage: boolean;
+  loadMore: () => void;
+  frontendChannels: Array<string>;
```
Comment (rohitmalhotra1420) [inline src/modules/channels/components/AllChannelsList.tsx]:
```suggestion
  suggestedChannels: Array<string>;
```

File: src/modules/channels/components/FrontendChannelListItem.tsx [hunk 13]
```diff
@@ -0,0 +1,20 @@
+import { useGetChannelDetails } from 'queries';
+import { FC } from 'react';
+import { AllChannelsListItem } from './AllChannelsListItem';
+
+export type FrontendChannelListItemProps = {
+  channelAddress: string;
+};
+
+const FrontendChannelListItem: FC<FrontendChannelListItemProps> = ({ channelAddress }) => {
```
Comment (rohitmalhotra1420) [inline src/modules/channels/components/FrontendChannelListItem.tsx]:
```suggestion
const SuggestedChannelListItem: FC<FrontendChannelListItemProps> = ({ channelAddress }) => {
```

**Review by rohitmalhotra1420 — APPROVED** [review]

**Pull request merged**